integration-tests/utils/integration-helpers.js

```diff
@@ -122,18 +131,52 @@ const deleteWorkspace = withSignedInPage(async ({ page, billingProject, workspac
   }
 });
 
-const withWorkspace = (test) => async (options) => {
-  console.log('withWorkspace ...');
-  const workspaceName = await makeWorkspace(options);
+const deleteWorkspaceV2 = withSignedInPage(async ({ page, billingProject, workspaceName }) => {
+  try {
+    await page.evaluate(
+      async (name, billingProject) => {
+        await window.Ajax().Workspaces.workspaceV2(billingProject, name).delete();
+      },
+      workspaceName,
+      billingProject
+    );
+    console.info(`Deleted workspace: ${workspaceName}`);
+  } catch (e) {
+    console.error(`Failed to delete workspace: ${workspaceName} with billing project: ${billingProject}`);
+    console.error(e);
+    throw e;
+  }
+});
+
+const withGcpWorkspace = (test) => async (options) => {
+  console.log('withGcpWorkspace ...');
+  const { workspaceName } = await makeGcpWorkspace(options);
 
   try {
     await test({ ...options, workspaceName });
   } finally {
-    console.log('withWorkspace cleanup ...');
+    console.log('withGcpWorkspace cleanup ...');
     await deleteWorkspace({ ...options, workspaceName });
   }
 };
 
+const withAzureWorkspace = (test) => async (options) => {
+  console.log('withAzureWorkspace ...');
+  options = {
+    ...options,
+    billingProject: options.billingProjectAzure,
+  };
+  const { workspaceName } = await makeWorkspace(options);
+  console.log(`withAzureWorkspace made workspace ${workspaceName}`);
+
+  try {
+    await test({ ...options, workspaceName });
+  } finally {
+    console.log('withAzureWorkspace cleanup ...', options, workspaceName);
```

Comment:
Can you not print the options? It adds a lot of text to the log.

Reply:
Thanks, thought I removed these :)